vortex-btrblocks/src/string.rs

```diff
@@ -17,28 +19,35 @@ use crate::{
 #[derive(Clone, Debug)]
 pub struct StringStats {
     src: VarBinViewArray,
+    exact_distinct_count: Option<u32>,
     estimated_distinct_count: u32,
     value_count: u32,
-    // null_count: u32,
+    null_count: u32,
 }
 
 /// Estimate the number of distinct strings in the var bin view array.
+/// If all are inlined then is count is exact and true is returned.
 #[allow(clippy::cast_possible_truncation)]
-fn estimate_distinct_count(strings: &VarBinViewArray) -> u32 {
+fn estimate_distinct_count(strings: &VarBinViewArray) -> (u32, bool) {
     let views = strings.views();
     // Iterate the views. Two strings which are equal must have the same first 8-bytes.
     // NOTE: there are cases where this performs pessimally, e.g. when we have strings that all
     // share a 4-byte prefix and have the same length.
     let mut distinct = HashSet::with_capacity(views.len() / 2);
+    let mut all_inlined = true;
     views.iter().for_each(|&view| {
-        let len_and_prefix = view.as_u128() as u64;
+        let len_and_prefix = view.as_u128();
         distinct.insert(len_and_prefix);
+        all_inlined &= view.is_inlined();
```

Comment:
We could instead to check if all views are inline and equal to the first one to check for constant?

Reply:
my thinking over the weekend is that it might be worth it to just use `is_constant` here, its pretty good and already has all kind of optimizations for `VarBinViewArray`.

Reply:
Might be worth benchmarking?